collector.go

```diff
@@ -123,6 +138,10 @@ func (cc *ChunkedCollector) Flush() error {
 		if err := cc.Collector.Collect(spanID, p...); err != nil {
```

Comment:
There is no guarantee on how long `Collect` can take, because the implementation is not concrete (it's an interface). I've updated the FlushTimeout docstrings to reflect this.

The assumption is that a Collect is inherently a small operation, <1s or so, and on avg <20ms or something like that.

The problem is RemoteCollector, which does not respect this, and can attempt reconnection after being disconnected from a broken Appdash collection server. If this is the case, and Collect does take longer than `FlushTimeout` due to a reconnection, then the remaining flush operation will be aborted -- but we're not doing anything fancy here / breaking the reconnection etc.

Yes, the flush operation might exceed `FlushTimeout` due to a reconnection taking a short while (50ms-2s-etc), for example, but we don't care about that as long as the rest of the queue is dropped (because doing so over and over again could take _minutes_ and cause the queue to grow and grow forever).